core/src/main/java/kafka/server/share/SharePartition.java

```diff
@@ -2189,12 +2188,16 @@ long lastOffset() {
             return lastOffset;
         }
 
-        // Visible for testing.
-        RecordState batchState() {
+        private InFlightState inFlightState() {
```

Comment:
nit: move private methods in the end i.e. after public -> protected -> default -> private